src/Sulu/Bundle/AdminBundle/Resources/js/stores/ResourceStore/ResourceStore.js

```diff
@@ -232,13 +232,22 @@ export default class ResourceStore {
             throw new Error('Copying from another locale does only work for objects with locales!');
         }
 
+        const destinationLocale = this.locale.get();
+
         return ResourceRequester
             .post(
                 this.resourceKey,
                 {},
-                {...options, action: 'copy-locale', dest: this.locale.get(), id: this.id, locale}
-            ).then(action((response) => {
+                {...options, action: 'copy-locale', dest: destinationLocale, id: this.id, locale}
+            ).then(action(() => {
+                // api will return data of there source locale, therefore we need to load the data of the new locale
+                return ResourceRequester.get(
```

Comment:
Making another request feels like some overhead here... Can't we fix that in another way? I know that the API returns the source locale, because when you copy the locale with the edit menu from the toolbar you can also copy it to multiple localizations, and then it would not be clear which one to return...

The reason I am worrying about this, is that sending too many requests was probably one of the main reasons the old administration felt much slower...

What if we change the API for the `copy-locale` action in a way, that let's the caller decide in which locale the content is returned? The current request looks like this, and it would probably be wise to keep this one working in the same way:

```
/admin/api/pages/d3354303-a11b-443f-9cb7-cace67a4e57c?webspace=sulu-test&action=copy-locale&dest=de&locale=en
```

In this example the english version would be returned from the API, although the german version was the target. This is driven by the `locale` query parameter. So we should probably keep the behavior, that the `locale` defines which locale is returned in the response.

An idea that just came to my mind would be that we introduce another query parameter, which is called `src` (opposed to `dest`). This query parameter could then be used to define from where the content is copied, and the `dest` parameter can be used to determine the target of the copy process. The the `locale` would be free to decide in which locale the content will be returned.

In order to keep BC, we can say that the `src` parameter is optional, and the value always falls back to the `locale` value. So the URL from above would still work, but at the same time something like this would be possible, that would save us the request created in this PR:

```
/admin/api/pages/d3354303-a11b-443f-9cb7-cace67a4e57c?webspace=sulu-test&action=copy-locale&src=en&dest=de&locale=de
```

WDYT?

Reply:
Yes, I think adjusting the API like you described would be the better solution.
Unfortunately, I dont have time to implement this at the moment, at least not this week. So it would be nice if someone else could take over here 😇  @sulu/core-team